graspy/subgraph/sg.py

```diff
@@ -0,0 +1,181 @@
+
+    Parameters
+    ----------
+    graphs: array-like, shape (n_vertices, n_vertices, s_samples)
+        A series of labeled (n_vertices, n_vertices) unweighted graph samples. If undirected, the upper or lower triangle matrices should be used.
+        
+    labels: vector, length (s_samples)
+        A vector of class labels. There must be a maximum of two classes.
+         
+    Attributes
+    ----------
+    contmat: array-like, shape (n_vertices, n_vertices, 2, 2)
+        An array that stores the 2-by-2 contingency matrix for each point in the graph samples.
+    sigsub: tuple, shape (2, n_edges)
+        A tuple of a row index array and column index array, where n_edges is the size of the signal-subgraph determined by *constraints*.
+    
+    References
+    ----------
+    .. [1] J. T. Vogelstein, W. R. Gray, R. J. Vogelstein, and C. E. Priebe, "Graph Classification using Signal-Subgraphs: Applications in Statistical Connectomics," arXiv:1108.1427v2 [stat.AP], 2012.
+    
+    """
+
+    def __init__(self, graphs, labels):
+        if not isinstance(graphs, np.ndarray):
+            msg = "Input array 'graphs' must be np.ndarray, not {}.".format(
+                type(graphs)
+            )
+            raise TypeError(msg)
+        if not isinstance(labels, (list, np.ndarray)):
+            msg = "Input vector 'labels' must be list or np.ndarray, not {}.".format(
+                type(labels)
+            )
+            raise TypeError(msg)
+
+        shape = np.shape(graphs)
+        if len(shape) != 3:
+            msg = "Input array 'graphs' must be 3-dimensional with shape (n_vertices, n_vertices, s_samples)."
+            raise ValueError(msg)
+        if shape[0] != shape[1]:
+            msg = "Input array 'graphs' must have matching number of vertices."
+            raise ValueError(msg)
+
+        if len(np.shape(labels)) != 1:
+            msg = "Input vector 'labels' must be 1-dimensional."
+            raise ValueError(msg)
+        if len(np.unique(labels)) > 2:
+            msg = "Input arrays must have a maximum of two classes, not {}.".format(
+                len(np.unique(labels))
+            )
+            raise ValueError(msg)
+        if len(labels) != shape[2]:
+            msg = "Input vector length must match the number of graph samples."
+            raise ValueError(msg)
+        else:
+            self.graphs = graphs
+            self.labels = labels
+
+    def __construct_contingency(self):
+        nverts = np.shape(self.graphs)[0]
+        out = np.zeros((nverts, nverts, 2, 2))
+        rowsum1 = sum(self.labels)
+        rowsum0 = len(self.labels) - rowsum1
+        for i in range(nverts):
+            for j in range(nverts):
+                a = sum(self.graphs[i, j, self.labels == 0])
+                b = sum(self.graphs[i, j, :]) - a
+                out[i, j, :, :] = [[a, rowsum0 - a], [b, rowsum1 - b]]
+        self.contmat = out
+
+    def fit(self, constraints):
```

Comment:
When you put graphs and labels here as inputs, I would call them `X` and `y` for consistency with all other classes.